src/Entity/IdeasWorkshop/BaseComment.php

```diff
@@ -126,6 +128,21 @@ public function enable(): void
         $this->enabled = true;
     }
 
+    /**
+     * @Assert\Callback(groups={"write"})
+     */
+    public function validateAuthorCgu(ExecutionContextInterface $context): void
+    {
+        if (!$this->getAuthor()->isCommentsCguAccepted() && $this->getAuthor()) {
```

Comment:
```php
!$this->getAuthor()->isCommentsCguAccepted() && $this->getAuthor()
```
here the second check is useless. If `$this->getAuthor()` returns null it will fail on first check because: `NULL->isCommentsCguAccepted()`.

this is a good order: 
```php
$this->getAuthor() && ...
```